### DISCUSSION TITLE
Include only a selected set of dependencies from upstream repository in my own update site

### DISCUSSION THREAD

**Discussion by jpfeuffer** [eclipse-tycho/tycho#2193] [Q&A] [answered]
Hi,

I have the following category.xml:

```
<?xml version="1.0" encoding="UTF-8"?>
<site>
   <feature id="mytool.dependency.feature" version="1.1.0"/>
   <feature id="mytool.feature" version="1.0.0"/>
</site>
```

and the following snippet in my parent pom.xml that includes the folder with the category.xml as module:

```
<repositories>
  <repository>
  <id>upstream</id>
  <layout>p2</layout>
  <url>${upstream.update.site}</url>
  </repository>
</repositories>
```

This repository contains "mytool.dependency.feature" and it is found correctly during build of mytool.feature which has "mytool.dependency.feature" as a dependency. Also `tycho-p2-repository-plugin:2.7.5:assemble-repository` for my update site shows:
`Fetching mytool.dependency.feature_1.0.0.jar from https://upstream.update.site/features/ (13.19kB)`

But it does not show up on the update site that is generated based on my category.xml. My complete example is a bit bigger than this but am I doing it right in theory or am I missing something important?
Note that mytool has even more dependencies but I do NOT want to include all of the transitive dependencies with e.g. includeAllDependencies. Just this selected feature.

**Comment by laeubi** [comment]
Repositories defined in the pom can not reference the current build, but its a bit unclear what you are trying to archive, can you 
provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue?

**Reply by jpfeuffer** [reply]
I kind of want to "re-export" one feature from an upstream p2 repository into a p2 repository (my-updatesite) that contains my-feature plus the one from the upstream repository. Preferably (see my initial attempt) this should be done in the build of my-feature and my-updatesite.

Hmm I will see what I can do but not sure when I will get to it. In the meantime, I just host two repositories.

**Reply by laeubi** [reply]
Its hard to tell why / what you see without an example but in general it should work as you have outlined ...

**Comment by jpfeuffer** [comment, marked as answer]
I can answer myself now. It was a version issue. The actual version of the feature I wanted to include was "1.1.0.somequalifier".
I was assuming that version matching works similarly as in the MANIFESTs, where (if I am not mistaken) if you specify a version number it
is interpreted as the minimum version. Here it seems like the version must match exactly? Anyway, I completely removed the version restriction, since I am sure that upstream always contains the correct version and it works now.
